postgres/src/main/java/org/jdbi/v3/postgres/HStoreArgumentFactory.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.postgres;
+
+import org.jdbi.v3.core.StatementContext;
+import org.jdbi.v3.core.argument.Argument;
+import org.jdbi.v3.core.argument.ArgumentFactory;
+
+import java.lang.reflect.Type;
+import java.sql.Types;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * An argument factory which binds Java's {@link Map} to Postgres' hstore type.
+ */
+public class HStoreArgumentFactory implements ArgumentFactory {
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Optional<Argument> build(Type type, Object value, StatementContext ctx) {
+        if (value instanceof Map) {
+            // Unfortunately, the PostgreSQL driver doesn't support mapping from Java maps to the hstore type.
+            // Therefore, we convert the map to a string representation and pass it to a PostgreSQL server.
+            return Optional.of((i, p, cx) -> p.setObject(i, toString((Map<String, ?>) value), Types.OTHER));
```

Comment:
The Postgres driver supports `hstore` out of the box. You can just call `setObject` with the map:

    statement.setObject(col, map);

Also the Postgres driver thoroughly escapes keys and values--see `HStoreConverter` in the Postgres JDBC jar.

Alternatively you could use `PGobject`:

    PGobject obj = new PGobject();
    obj.setType("hstore");
    obj.setValue(HStoreConverter.toString(map));
    statement.setObject(col, obj);